### ISSUE TITLE
How to make styles available in JS using Rollup?

### ISSUE THREAD

**Issue by morewry** [tivac/modular-css#451]
When using [postcss-loader](https://github.com/postcss/postcss-loader) for [webpack](https://webpack.js.org/), importing a [CSS Module](https://github.com/css-modules/css-modules) gives your script both a map of the scoped class names and the postprocessed styles. So you can, for example:

###### theme.css
```
@value textColor: red;
```

###### styles.css
```
@value textColor from "./theme.css";
.className { color: textColor; }
```

###### component.js
```javascript
import styles from './styles.css';

console.log(styles.toString());
// the post-processed styles, e.g.
// .className { color: red; }

console.log(styles.locals.className);
// the scoped class name, e.g.
// styles_className_8348jdfkasfdj
```

I need both of these for my use case ([Shadow DOM](https://dom.spec.whatwg.org/#shadowroot)) and I can't figure out how to configure this plugin to get that result. 

I have gotten as far as getting the postprocessed styles in an external CSS file and the map of named exports in my JS, but I don't see how to get the postprocessed styles available in my JS for me to output inside my component's [Shadow root](https://dom.spec.whatwg.org/#shadowroot).

**Comment by morewry** [comment]
I'm trying that out locally (had to drill into `result.details.result.css`, though) and I'll try to PR it, but I don't have very much time to spare--I'm trying to get something to production, but am kind of stuck in the middle of some rocks and some hard places with different build options. That being said, your suggestion works well for me and allows me to move the direction I'd prefer, so I'd like to rely on it and contribute it back to the project in a way that can be confidently merged and released.

What about adding an option for including the `styles`? 

I think that CSS Modules should really define the interface when a CSS file is imported as a CSS Module, so that you don't have to change your code depending on the bundler/plugin that you're using. For example, using Webpack & css-loader, you use `mystyles.locals[className]` & `mystyles.toString()` where using different tools means you use `mystyles[className]` & `mystyles.css` (or some other small shift that nevertheless changes how you work with the imported styles). I think any such interface would have to include a way to access the post-processed styles themselves. 

So it's not that I think it's a bad idea to always include the `styles`, but the `after` and `done` hooks are the ones you intend to be used to run other transformations on the styles, is that correct? If that is the case, and a user is doing a transformation in the `done`, the `styles` in the JS wouldn't have that transformation. I'm assuming that's the footgun to which you are referring? If so, this being a new feature for this project, I think it makes sense to hide it behind an option and document that using this option and the `done` hook together isn't recommended for that reason and that, instead of running cssnano in `done` with this option, you should run it in `after`.

Assuming that you could successfully use autoprefix and cssnano in `after` and that it's just not always _best_ place, depending on use case. If not, it doesn't seem like a safe feature to add.

**Comment by morewry** [comment]
Another thing I noticed is that I have inline source maps embedded in the styles. I haven't done any particular configuration for source maps, so I'm assuming that's a default behavior. For production builds I think many people would want those gone, so if there's anything special that needs doing in the PR to allow that...?

**Comment by morewry** [comment]
Any ideas as to why I am getting:

```
TypeError: parser.processSync is not a function
```

In many of the test suites after running `npm install` followed by `npm run test`?

**Comment by tivac** [comment]
Not without a trace, sorry! Make sure you also run `npx lerna exec -- npm i` so all the per-package dependencies get installed.

**Comment by tivac** [comment]
Sorry for asking you to do the legwork, btw. I just don't have any use-case for that kind of flow so I'm hesitant to build something I'm not going to use. If it's easier for you to simply make an explicit API/behavior proposal that lays out exactly what you want and how you'd configure `modular-css-rollup` (& the other bundlers) to achieve it I can probably dredge up some time to work on it.

Something w/ lots of code examples of configuration, input, and expected output ideally to make sure I don't misunderstand :)

**Comment by morewry** [comment]
No worries there, just giving a heads up I may hit a limit with what I can contribute. I have enough for a PR to get us started, at least.

**Comment by morewry** [comment]
Also, thanks for suggesting a PR and being so specific about the best place to start. It helped me to try it out and quickly get to a point where I was able to open this PR. That's a really nice thing to do for contributors. 

And the repo is well organized, too.

**Comment by tivac** [comment]
Released in `modular-css-rollup@14.1.0` by using the `styleExport` option!
